tests/nested/manual/core20-boot-config-update/task.yaml

```diff
@@ -0,0 +1,155 @@
+
+  echo "Leave a marker file that triggers boot config assets to be injected"
+  echo 'bootassetstesting' > snapd-snap/usr/lib/snapd/bootassetstesting
+
+  snap pack snapd-snap --filename=snapd-boot-config-update.snap
+  rm -rf snapd-snap
+
+  if [ "$VARIANT" != "no-gadget" ]; then
+    echo "Grab and prepare the gadget snap"
+    snap download --basename=pc --channel="20/edge" pc
+    unsquashfs -d pc-gadget pc.snap
+
+    echo "Sign the shim binary"
+    KEY_NAME=$(nested_get_snakeoil_key)
+    SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+    SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+    nested_secboot_sign_gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+
+    case "$VARIANT" in
+      gadget-extra)
+        echo 'snapd.debug=1 hello from test' > pc-gadget/cmdline.extra
+        ;;
+      gadget-full)
+        # keep the console so that we have useful debug logs
+        echo 'snapd.debug=1 console=ttyS0 full hello from test' > pc-gadget/cmdline.full
+        ;;
+      *)
+        echo "unexpected variant $VARIANT, fix the test"
+        exit 1
+        ;;
+    esac
+    snap pack pc-gadget/ extra-snaps/
+  fi
+
+  "$TESTSTOOLS"/nested-state build-image core
+  "$TESTSTOOLS"/nested-state create-vm core
+
+  nested_copy snapd-boot-config-update.snap
+
+debug: |
+  cat boot-chains-before.json || true
+  cat boot-chains-after.json || true
+
+execute: |
+  if [ "$TRUST_TEST_KEYS" = "false" ]; then
+      echo "This test needs a build of snapd with testing features enabled"
+      exit
+  fi
+
+  # shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  nested_exec sudo cat /var/lib/snapd/device/fde/boot-chains > boot-chains-before.json
+  SEALED_KEY_MTIME_1="$(nested_exec sudo stat --format="%Y" /run/mnt/ubuntu-boot/device/fde/ubuntu-data.sealed-key)"
+  RESEAL_COUNT_1="$(jq -r '.["reseal-count"]' < boot-chains-before.json )"
+  jq -r '.["boot-chains"][]["kernel-cmdlines"][]' < boot-chains-before.json | not MATCH ' bootassetstesting'
+  case "$VARIANT" in
+    gadget-extra)
+      jq -r '.["boot-chains"][]["kernel-cmdlines"][]' < boot-chains-before.json | MATCH 'snapd.debug=1 hello from test'
+      ;;
+    gadget-full)
+      jq -r '.["boot-chains"][]["kernel-cmdlines"][]' < boot-chains-before.json | MATCH ' full hello from test'
+      ;;
+    no-gadget)
+      jq -r '.["boot-chains"][]["kernel-cmdlines"][]' < boot-chains-before.json | not MATCH 'hello from test'
+      ;;
+  esac
+
+  echo "Install new (unasserted) snapd and wait for reboot/change finishing"
+  boot_id="$( nested_get_boot_id )"
+  REMOTE_CHG_ID=$(nested_exec sudo snap install --dangerous snapd-boot-config-update.snap --no-wait)
+  nested_exec sudo snap watch "${REMOTE_CHG_ID}"
+  if [ "$VARIANT" != "gadget-full" ]; then
+    # reboot is automatically requested by snapd only if part of the command
+    # line is changed, in case of gadget overriding the command line fully
+    # (gadget-full variant), there is no resealing and no reboot
```

Comment:
Maybe a small comment here *why* no resealing and no reboot?

Reply:
Tweaked the comment a bit.